src/lib/reasoners/rel_utils.ml

```diff
@@ -815,4 +849,44 @@ end = struct
     MX.fold (fun r _ acc ->
         f r acc
       ) c.args_map acc
+
+  let simplify_pending =
+    (* Recursion needed because adding new constraints changes the pending set
+       and they also need to be simplified *)
+    let rec simplify_aux eqs t to_simplify =
+      let eqs = ref eqs in
```

Comment:
We accumulate all the discovered equations by the simplification pass in this list but this list could contain duplications. We may construct a set and extract the list of its elements later?